src/PowerShellEditorServices/Workspace/Workspace.cs

```diff
@@ -624,6 +624,65 @@ private static string UnescapeDriveColon(string fileUri)
             return sb.ToString();
         }
 
+        /// <summary>
+        /// Converts a file system path into a DocumentUri required by Language Server Protocol.
+        /// </summary>
+        /// <remarks>
+        /// When sending a document path to a LSP client, the path must be provided as a
+        /// DocumentUri in order to features like the Problems window or peek definition
+        /// to be able to open the specified file.
+        /// </remarks>
+        /// <param name="path">
+        /// A file system path. Note: if the path is already a DocumentUri, it will be returned unmodified.
+        /// </param>
+        /// <returns>The file system path encoded as a DocumentUri.</returns>
+        public static string ConvertPathToDocumentUri(string path)
+        {
+            const string fileUriPrefix = "file:///";
+
+            if (path.StartsWith("untitled:", StringComparison.Ordinal))
+            {
+                return path;
+            }
+
+            if (path.StartsWith(fileUriPrefix, StringComparison.Ordinal))
+            {
+                return path;
+            }
+
+            string escapedPath = Uri.EscapeDataString(path);
+            var docUriStrBld = new StringBuilder(escapedPath);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // VSCode file URIs on Windows need the drive letter lowercase.
+                if (path.Contains(':'))
+                {
+                    for (int i = 1; i < docUriStrBld.Length - 2; i++)
```

Comment:
this code is a tad complex. Comments would be nice. For example, why do we start at 1 here?

Reply:
So we're avoiding another index check below because I need to access `i - 1` and if i is 0, that will throw an IndexOutOfRangeException.  Anyway, if the path starts with a `:` then then is no drive letter before it to lower case.  I'll add a comment.